manual_correction.py

```diff
@@ -638,64 +645,70 @@ def main():
                     # Include session in the filename
                     else:
                         other_contrast_filename = subject + '_' + ses + '_' + args.load_other_contrast + '.nii.gz'
-                    fname_other_contrast = os.path.join(utils.get_full_path(args.path_in), subject, ses, contrast,
-                                                        other_contrast_filename)
+                    fname_other_contrast = os.path.join(path_img, subject, ses, contrast, other_contrast_filename)
                 else:
                     fname_other_contrast = None
                 # Construct absolute path to the input label (segmentation, labeling etc.) file
                 # For example: '/Users/user/dataset/data_processed/sub-001/anat/sub-001_T2w_seg.nii.gz'
-                fname_seg = utils.add_suffix(fname, suffix_dict[task])
-                # Construct absolute path to the derivative file (i.e., path where manually corrected file will be saved)
-                # For example: '/Users/user/dataset/derivatives/labels/sub-001/anat/sub-001_T2w_seg-manual.nii.gz'
-                fname_label = os.path.join(path_out_deriv, subject, ses, contrast,
-                                           utils.add_suffix(utils.remove_suffix(filename, args.suffix_files_in),
-                                                            suffix_dict[task] + '-manual'))
-                # Create output folders under derivative if they do not exist
-                os.makedirs(os.path.join(path_out_deriv, subject, ses, contrast), exist_ok=True)
+                fname_label = utils.add_suffix(os.path.join(path_label, subject, ses, contrast, filename), suffix_dict[task])
+                
+                # Construct absolute path to the output file (i.e., path where manually corrected file will be saved)
+                # For example: '/Users/user/dataset/data_processed/sub-001/anat/sub-001_T2w_seg.nii.gz'
+                # The information regarding the modified data will be stored within the sidecar .json file
+                temp_fname_out = utils.add_suffix(os.path.join(path_out, subject, ses, contrast, filename), suffix_dict[task])
+                
+                # This condition is used to remove the suffix -manual from our current datasets (can be deleted later)
+                if '-manual' in temp_fname_out:
+                    fname_out = temp_fname_out.replace('-manual','')
+                else:
+                    fname_out = temp_fname_out
```

Comment:
Should I add a step at the end of the script to remove the original label if this one contains `-manual` and is inside the same folder ?

Reply:
I do not think so. Each user will be responsible for renaming already existing `-manual` files.